pyiron_base/project/maintenance.py

```diff
@@ -60,6 +67,42 @@ def get_repository_status():
         return report
 
 
+class UpdateMaintenance:
+    def __init__(self, project):
+        self._project = project
+
+    def base_v0_3_to_v0_4(self, project=None):
```

Comment:
I suggest we have a separate function called `base_to_latest()`, which upgrade the project to be compatible with the latest version. At the moment, such a function would refer to this function.

Reply:
Fair idea, right now this would work fine. However, we would need something like
```python
def base_to_latest(self, start_version, project=None):
  mayor, minor, patch = start_version.split('.')
  if mayor != 0:
    raise ValueError

  if minor < 4:
    base_v0_3_to_v0_4(project)
  # Hope we will not need this, though :)
  if minor < 7:
    base_v0_6_to_v07(project)
```

Reply:
and actually if we call it 'latest' we should also check if there is a more recent `pyiron_base` release available.

Reply:
> and actually if we call it 'latest' we should also check if there is a more recent `pyiron_base` release available.

If that bloats the logic to much, just name to `*_to_current` or something similar.  People already have conda/pip to check for new versions, no need for us to re-implement that imho.